src/pilgrim/utils/config_manager.py

```
@@ -0,0 +1,107 @@
+                    "auto_open_on_creation": False
+                }
+            }
+        }
+        if config is None:
+            config = default
+
+        try:
+            with open(f"{config_dir}/config.toml", "wb") as f:
+                tomli_w.dump(config, f)
+        except Exception as e:
+            raise RuntimeError(f"Error creating configuration: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)